nilearn/datasets/func.py

```diff
@@ -916,107 +923,104 @@ def fetch_localizer_contrasts(contrasts, n_subjects=None, get_tmaps=False,
+                        subject_id, contrast, data_type)])
+                if _is_valid_path(path, index, verbose=verbose):
+                    file_url = root_url.format(index[path][1:])
+                    opts = {"move": file_path}
+                    filenames.append((file_path, file_url, opts))
+                    files.setdefault(data_type, []).append(file_path)
+
     # Fetch masks if asked by user
     if get_masks:
-        urls.append("%sbrainomics_data_masks.zip?rql=%s&vid=data-zip"
-                    % (root_url,
-                       _urllib.parse.quote(base_query % {"types": '"boolean mask"',
-                                                  "label": "mask"},
-                                    safe=',()')))
         for subject_id in subject_ids:
             file_path = os.path.join(
                 "brainomics_data", subject_id, "boolean_mask_mask.nii.gz")
-            file_tarball_url = urls[-1]
-            filenames.append((file_path, file_tarball_url, opts))
+            path = "/".join([
+                "/localizer", "derivatives", "spm_1st_level",
+                "sub-%s" % subject_id, "sub-%s_mask.nii.gz" % subject_id])
+            if _is_valid_path(path, index, verbose=verbose):
+                file_url = root_url.format(index[path][1:])
+                opts = {"move": file_path}
+                filenames.append((file_path, file_url, opts))
+                files.setdefault("masks", []).append(file_path)
+
     # Fetch anats if asked by user
     if get_anats:
-        urls.append("%sbrainomics_data_anats.zip?rql=%s&vid=data-zip"
-                    % (root_url,
-                       _urllib.parse.quote(base_query % {"types": '"normalized T1"',
-                                                  "label": "anatomy"},
-                                    safe=',()')))
         for subject_id in subject_ids:
             file_path = os.path.join(
                 "brainomics_data", subject_id,
                 "normalized_T1_anat_defaced.nii.gz")
-            file_tarball_url = urls[-1]
-            filenames.append((file_path, file_tarball_url, opts))
-    # Fetch subject characteristics (separated in two files)
-    if url is None:
-        url_csv = ("%sdataset/cubicwebexport.csv?rql=%s&vid=csvexport"
-                   % (root_url, _urllib.parse.quote("Any X WHERE X is Subject")))
-        url_csv2 = ("%sdataset/cubicwebexport2.csv?rql=%s&vid=csvexport"
-                    % (root_url,
-                       _urllib.parse.quote("Any X,XI,XD WHERE X is QuestionnaireRun, "
-                                    "X identifier XI, X datetime "
-                                    "XD", safe=',')
-                       ))
-    else:
-        url_csv = "%s/cubicwebexport.csv" % url
-        url_csv2 = "%s/cubicwebexport2.csv" % url
-    filenames += [("cubicwebexport.csv", url_csv, {}),
-                  ("cubicwebexport2.csv", url_csv2, {})]
+            path = "/".join([
```

Comment:
Same: `os.path.join()`

Reply:
Will break on Windows, we have to keep "/".join().

Reply:
Ah, my wrong. These are URLs, not paths. Sorry.

Reply:
Maybe change the variable name to `url_path` or something better?

Reply:
Right. I would avoid mixing the two terms _url_ and _path_:
* use _path_ in names of variables that contain a path and use `os.path.join()` to manipulate them,
* use _url_ in names of variables that contain a URL and use `"/".join()` to manipulate them.

Therefore I would avoid _url_path_. Perhaps _file_url_?

Reply:
Sure, like I said, something better, that clearly communicates what it is and what it is not. I'll leave the choice of name to @AGrigis

Reply:
or urljoin
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

Reply:
Ah indeed, but is it worth adding a dependency on _urllib_ just for this? Perhaps we should leave it as is. Antoine won't have much more time to invest in preparing this patch. If needed I'm willing to add minor changes later on, just tell me what needs to be done. Not sure how to take over Antoine commits his changes.

Reply:
`urllib` is a standard python package. Not an external dependency.

Reply:
keeping it as it is is also fine